java/app/src/main/java/com/google/android/gms/samples/wallet/activity/CheckoutActivity.java

```diff
@@ -95,57 +102,124 @@ protected void onCreate(Bundle savedInstanceState) {
     possiblyShowGooglePayButton();
   }
 
+  /**
+   * Add a menu option to trigger a notification
+   */
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    MenuInflater inflater = getMenuInflater();
+    inflater.inflate(R.menu.menus, menu);
+    return true;
+  }
+
+  /**
+   * Handle selection in the options menu
+   */
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case R.id.send_notification:
+        Notifications.triggerPaymentNotification(this);
+        return true;
+      default:
+        return super.onOptionsItemSelected(item);
+    }
+  }
+
+  /**
+   * Handle a resolved activity from the Google Pay payment sheet.
+   *
+   * @param requestCode Request code originally supplied to AutoResolveHelper in requestPayment().
+   * @param resultCode  Result code returned by the Google Pay API.
+   * @param data        Intent from the Google Pay API containing payment or error data.
+   * @see <a href="https://developer.android.com/training/basics/intents/result">Getting a result
+   * from an Activity</a>
+   */
+  @Override
+  public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    switch (requestCode) {
+      // value passed in AutoResolveHelper
+      case LOAD_PAYMENT_DATA_REQUEST_CODE:
+        switch (resultCode) {
+
+          case Activity.RESULT_OK:
+            PaymentData paymentData = PaymentData.getFromIntent(data);
+            handlePaymentSuccess(paymentData);
+            break;
+
+          case Activity.RESULT_CANCELED:
+            // The user cancelled the payment attempt
+            break;
+
+          case AutoResolveHelper.RESULT_ERROR:
+            Status status = AutoResolveHelper.getStatusFromIntent(data);
+            handleError(status.getStatusCode());
+            break;
+
+          default:
+            // Do nothing.
```

Comment:
If it's going to be empty, better delete it

Reply:
Good catch